mcf-browser/src/back-end/graph.js

```diff
@@ -0,0 +1,318 @@
+   * already exists. If remote node exists, then the remote node is absorbed by
+   * current node via mergeNode() method.
+   *
+   * @param {string} dcid The dcid to be added to the Node object.
+   */
+  setDCID(dcid) {
+    const remote = Node.nodeHash[dcid];
+    if (remote) {
+      this.mergeNode(remote);
+    }
+    this.dcid = dcid;
+    Node.nodeHash[dcid] = this;
+  }
+
+  /**
+   * Moves the assertions and inverse Assertions from the given param node to
+   * the calling Node object by changing the src property for assertions and the
+   * target property of the invAssertions.
+   *
+   * @param {Node} absorbedNode The node object whose triples should be copied.
+   */
+  mergeNode(absorbedNode) {
+    if (this.localId === absorbedNode.localId) {
+      return;
+    }
+
+    absorbedNode.getAssertions().forEach((assert) => {
+      assert.src = this;
+      assert.nextAssertion = this.assertions;
+      this.assertions = assert;
+    });
+
+    absorbedNode.getInvAssertions().forEach((invAssert) => {
+      invAssert.target = this;
+      invAssert.invNextAssertion = this.invAssertions;
+      this.invAssertions = invAssert;
+    });
+  }
+
+  /**
+   * Sets the property existsInKG to true if the Node has triples in the DC KG.
+   */
+  async setExistsInKG() {
+    if (!this.dcid || this.existsInKG) {
+      return;
+    }
+    this.existsInKG = await getExistsInKG(this.dcid);
+  }
+
+  /**
+   * Creates Assertion objects from a list of property labels by calling the
+   * helper function getRemotePropertyValues from utils.js to find the values
+   * in Data Commons given the current node, a property label, and the direction
+   * of the label.
+   *
+   * @param {Array<string>} propLabels List of property labels associated with
+   *     the calling Node object in Data Commons.
+   * @param {boolean} isInverse True if the list of labels are incoming labels,
+   *     meaning the calling Node object is the target of the triple. False if
+   *     the calling Node is the source of the triple.
+   */
+  async createAssertionsFromLabels(propLabels, isInverse) {
+    for (const label of propLabels) {
+      await getRemotePropertyValues(this.dcid, label, isInverse)
+          .then((valueList) => {
+            if (!valueList) {
+              throw new Error('No property values for dcid: ' + this.dcid +
+                              ' label: ' + label);
+            }
+
+            valueList.forEach((valueObj) => {
+              const val = getValueFromValueObj(valueObj);
+              const source = isInverse ? val : this;
+              const target = isInverse ? this : val;
+              new Assertion(source, label, target, valueObj.provenanceId);
+            });
+          });
+    }
+  }
+
+  /**
+   * Stores remote triples as assertions and inverse Assertions of the calling
+   * Node object. Sets the alreadyFetched property to true if data is fetched.
+   */
+  async fetchRemoteData() {
```

Comment:
to confirm, the caller is not in this PR?

Reply:
yes, this is called from the 'front-end" side. I have not submitted a PR for that code yet.